opm/autodiff/FlowMainEbos.hpp

```diff
@@ -541,13 +568,51 @@ namespace Opm
         {
             bool output      = param_.getDefault("output", true);
             bool output_ecl  = param_.getDefault("output_ecl", true);
-            const Grid& grid = this->grid();
-            if( output && output_ecl && output_cout_)
+            if( output && output_ecl )
             {
-                const EclipseGrid& inputGrid = eclState().getInputGrid();
-                eclipse_writer_.reset(new EclipseWriter(eclState(), UgGridHelpers::createEclipseGrid( grid , inputGrid )));
-                eclipse_writer_->writeInitial(geoprops_->simProps(grid),
-                                              geoprops_->nonCartesianConnections());
+                const Grid& grid = this->globalGrid();
+
+                if( output_cout_ ){
+                    const EclipseGrid& inputGrid = eclState().getInputGrid();
+                    eclipse_writer_.reset(new EclipseWriter(eclState(), UgGridHelpers::createEclipseGrid( grid , inputGrid )));
+                }
+
+                const NNC* nnc = &geoprops_->nonCartesianConnections();
+                data::Solution globaltrans;
+
+                if ( must_distribute_ )
+                {
+                    // dirty and dangerous hack!
+                    // We rely on opmfil in GeoProps being hardcoded to true
+                    // which prevents the pinch processing from running.
+                    // Ergo the nncs are unchanged.
+                    nnc = &eclState().getInputNNC();
+
+                    // Gather the global simProps
+                    data::Solution localtrans = geoprops_->simProps(this->grid());
+                    for( const auto& localkeyval: localtrans)
+                    {
+                        auto& globalval = globaltrans[localkeyval.first].data;
+                        const auto& localval  = localkeyval.second.data;
+
+                        if( output_cout_ )
+                        {
+                            globalval.resize( grid.size(0));
+                        }
+                        gatherCellDataToGlobalIterator(this->grid(), localval.begin(),
+                                                       globalval.begin());
+                    }
+                }
+                else
+                {
+                    globaltrans = geoprops_->simProps(grid);
```

Comment:
isn't `simProps()` an awkward name for a method that returns the the transmissibilites mapped to the logically Cartesian grid?